File: FrontendSilky/src/app/shared/models/purchaseitem.ts
import { Product } from "./product";

export class PurchaseItem {

  public id: number;
  public price: number;
  public product: Product;
  public quantity: number;


  public get _id(): number {
    return this.id;
  }
  public set _id(value: number) {
    this.id = value;
  }


  public get _price(): number {
    return this.price;
  }
  public set _price(value: number) {
    this.price = value;
  }


  public get _product(): Product {
    return this.product;
  }
  public set _product(value: Product) {
    this.product = value;
  }

  public get _quantity(): number {
    return this.quantity;
  }
  public set _quantity(value: number) {
    this.quantity = value;
  }
}

File: BackendSilky/src/main/java/com/proweb/Silky/model/PurchaseItem.java
package com.proweb.Silky.model;

import java.math.BigDecimal;

import javax.persistence.Basic;
import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "purchase_items")
public class PurchaseItem {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;
	@Basic
	private BigDecimal price;
	@ManyToOne(fetch=FetchType.EAGER,cascade = CascadeType.REMOVE)
	private Product product;
	@Basic
	private int quantity;
	public long getId() {
		return id;
	}
	public void setId(long id) {
		this.id = id;
	}
	public BigDecimal getPrice() {
		return price;
	}
	public void setPrice(BigDecimal price) {
		this.price = price;
	}
	public Product getProduct() {
		return product;
	}
	public void setProduct(Product product) {
		this.product = product;
	}
	public int getQuantity() {
		return quantity;
	}
	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}
	@Override
	public String toString() {
		return "PurchaseItem [id=" + id + ", price=" + price + ", product=" + product + ", quantity=" + quantity + "]";
	}

	
	

}
